consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/PostMergeContext.java

```diff
@@ -265,4 +269,14 @@ private static class PayloadTuple {
       this.block = block;
     }
   }
+
+  public PostMergeContext setIsNearHeadCheckpointSync(final boolean isNearHeadCheckpointSync) {
+    this.isNearHeadCheckpointSync = isNearHeadCheckpointSync;
+    return this;
```

Comment:
nit: is this necessary since we're not in a builder pattern?

Reply:
There are some methods that implement in the same way probably to make use of the method chaining feature. I've implemented this way to not break the pattern we use in the MergeBesuControllerCuilder